helix-core/src/test/java/org/apache/helix/manager/zk/TestZkBaseDataAccessor.java

```diff
@@ -188,6 +191,40 @@ public void testSyncCreate() {
     System.out.println("END " + testName + " at " + new Date(System.currentTimeMillis()));
   }
 
+  @Test
+  public void testSyncCreateWithCustomSerializer() {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String testName = className + "_" + methodName;
+
+    System.out.println("START " + testName + " at " + new Date(System.currentTimeMillis()));
+
+    String path = String.format("/%s/%s", _rootPath, "msg_0");
+
+    ZkBaseDataAccessor<ZNRecord> accessor = new ZkBaseDataAccessor<>(_gZkClient);
+
+    String recordData = "DummyContent";
+    ZkSerializer zkSerializer = new ZkSerializer() {
+      @Override
+      public byte[] serialize(Object o)
+          throws ZkMarshallingError {
+        String st = (String) o;
+        return st.getBytes();
+      }
+
+      @Override
+      public Object deserialize(byte[] bytes)
+          throws ZkMarshallingError {
+        return bytes;
+      }
+    };
+    boolean success = accessor.create(path, recordData, AccessOption.PERSISTENT, zkSerializer);
+    Assert.assertTrue(success);
+    Object data = accessor.get(path, null, AccessOption.PERSISTENT, zkSerializer);
+    Assert.assertEquals(new String((byte[]) data), "DummyContent");
+    System.out.println("END " + testName + " at " + new Date(System.currentTimeMillis()));
+  }
+
```

Comment:
Could we also add a case for get()?

Reply:
The simple get test is already verified after creation, do you think is it still needed for adding one specific case for get?

Reply:
In that case, I'm fine with not adding a test case for get.